contracts/contracts/GameItems.sol

```diff
@@ -1,32 +1,223 @@
+        uint256 _currentItemIndex = numberItems;
+        metadata[_currentItemIndex] = newItem;
+        _setURI(_currentItemIndex, tokenURI);
+        numberItems++;
+        emit NewItem(_currentItemIndex);
+    }
+
+    /// @notice Function takes a tokenId and checks if the signer has the required building blocks
+    /// @dev Uses the Storage Metadata for the checks to be fair across the board
+    /// @dev Balance Must be Greater than Or Equal to One
+    /// @dev Emits [Combined] Event if Combined
+    /// @param newItemId tokenId the signer is attempting to create
+    function combine(uint256 newItemId) override external {
+        require(newItemId <= numberItems, ITEM_DOES_NOT_EXIST);
+        GameItemMetadata storage item = metadata[newItemId];
+        for (uint256 i = 0; i < item.combined.length; i++) {
+            require(balanceOf(msg.sender, item.combined[i]) >= 1, USER_NO_ITEM);
+        }
+        _internalMint(msg.sender, newItemId);
+        emit Combined(msg.sender, newItemId, block.timestamp);
+    }
 
-    constructor(address initialOwner) ERC1155("EHCCContest") {
-        _setupRole(ADMIN_ROLE, initialOwner);
+    /// @notice Returns on Chain Token - Single (Metadata)
+    function getOnChainToken(uint256 tokenId)  external view  returns (GameItemMetadata memory) { 
+        return metadata[tokenId];
     }
 
-    function itemDetails(uint tokenID) public returns (GameItemMetadata memory) {
-      // return the metadata for the item
-    };
+    /// @notice Loads all the Metadata by a player in order to allow them to load the data
+    /// @dev Thoughts on making this a signed request during ETH CC?
+    function getItems(address _address) override external view returns (GameItemMetadata[] memory) {
+        GameItemMetadata[] memory _items = new GameItemMetadata[](numberItems);
+        for (uint256 i = 0; i < _items.length; i++) {
+            if (balanceOf(_address, i) <= 1) {
+                _items[i] = metadata[i];
+            }
+        }
 
-    /**
-     * @dev See {IERC165-supportsInterface}.
-     */
-    function supportsInterface(bytes4 interfaceId)
-        public
-        view
-        virtual
-        override(AccessControl, ERC1155)
-        returns (bool)
-    {
+        return _items;
+    }
+
+    /// @notice Retreives the current number of players/holders
+    function getNumberPlayers() override external view returns (uint256) {
+        return numberPlayers;
+    }
+
+    /// @notice Retreives Unlock Date
+    /// @dev Can be deprecated
+    /// @return uint256 time in epoch * 1000
+    function getUnlockDate() external view returns (uint256) {
+        return unlockDate;
+    }
+
+    function initialMint(address receiver) override external onlyMinter {
+        //// TESTING VALUE === 6
+        require(numberItems == 6, CONTRACT_IN_ACTIVE); /// PRODUCTION VALUE === require(numberItems == 12, CONTRACT_IN_ACTIVE);
```

Comment:
Not sure I understand this?

Reply:
It's mostly just a mock piece right now for testing. 
It technically is not necessary, was mostly for testing in order to confirm that minting could not happen without a certain number of items to confirm that the contract stored them in the proper manner. That can be commented or deleted entirely.